de-lib/src/main/java/org/iplantc/de/desktop/client/views/windows/WindowBase.java

```diff
@@ -618,4 +627,59 @@ public void logWindowCloseToIntercom(WindowConfig config) {
         }
     }
 
+    public abstract String getWindowType();
+
+    public abstract FastMap<String> getAdditionalWindowStates();
+
+    public void restoreWindowState() {
+        if (ws.isMaximized()) {
+            Scheduler.get().scheduleDeferred((Command)() -> maximize());
+        } else if (ws.isMinimized()) {
+            Scheduler.get().scheduleDeferred((Command)() -> minimize());
+        } else {
+            int left = ws.getWinLeft();
+            int top = ws.getWinTop();
+            setPagePosition(left, top);
+        }
+    }
+
+    @Override
+    public WindowState getWindowStateFromLocalStorage(String tag) {
+        WindowState ws = wsf.windowState().as();
+        String prefix = WebStorageUtil.LOCAL_STORAGE_PREFIX + getWindowType();
+        String suffix = "#" + tag + "#" + userInfo.getUsername();
+        String height = WebStorageUtil.readFromStorage(prefix + WindowState.HEIGHT + suffix);
```

Comment:
The prefix and suffix  is needed (or a one longPrefix + var) is needed to restore the window state correctly.  But I had the same feeling this that piece of code doesn't belong here. Anyway I have a wrapper to store and retrieve `WindowState`. This will also take of type conversion. Unit test have added / updated accordingly.